### PR TITLE
spirv-headers: 1.5.3 -> 2020-01-27

### PR DESCRIPTION
**Pull request by jansol**
###### Motivation for this change

New upstream release

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ralith — APPROVED** [review]

File: pkgs/development/libraries/spirv-headers/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "spirv-headers";
-  version = "1.5.3";
+  version = "1.5.4.raytracing.fixed";
 
   src = fetchFromGitHub {
     owner = "KhronosGroup";
     repo = "SPIRV-Headers";
     rev = version;
-    sha256 = "069sivqajp7z4p44lmrz23lvf237xpkjxd4lzrg27836pwqcz9bj";
+    hash = "sha256:WKDiLHSe5v2Ieh9Q1nad4lgvTHY6Ji5PkdJkxXAV94k=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/spirv-headers/default.nix:11]:
```suggestion
    sha256 = "WKDiLHSe5v2Ieh9Q1nad4lgvTHY6Ji5PkdJkxXAV94k=";
```

Comment (jansol) [reply pkgs/development/libraries/spirv-headers/default.nix:11]: Is there some general convention around this? I have seen both variants used

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114366` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>playonlinux: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>spirv-tools: <a href="https://termbin.com/l438">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/64473d5213b18a32b599f319a7a4d6ec/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>pipelight</li>
<li>spirv-headers</li>
<li>vkBasalt</li>
<li>vulkan-tools-lunarg</li>
<li>vulkan-validation-layers</li>
</ul>
</details>

**Comment by jansol** [comment]
Updating to untagged rev that is specified in the DEPS file of spirv-tools instead since spirv-tools does not build with the last tagged one.

File: pkgs/development/libraries/spirv-headers/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/spirv-headers/default.nix]: No, not really. Both should be fine but ``sha256 = "sha256:`` shouldn't be used.

File: pkgs/development/libraries/spirv-headers/default.nix [hunk 2]
```diff
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "spirv-headers";
-  version = "1.5.3";
+  version = "2020-01-27";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/spirv-headers/default.nix:5]:
```suggestion
  version = "unstable-2020-01-27";
```

File: pkgs/development/libraries/spirv-headers/default.nix (hunk 1 above)
Comment (jansol) [reply pkgs/development/libraries/spirv-headers/default.nix]: That's not what is here though?

**Comment by SuperSandro2000** [comment]
fails to build:

```
Scanning dependencies of target SPIRV-Tools
[  4%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/util/bit_vector.cpp.o
[  5%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/util/string_utils.cpp.o
[  5%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/assembly_grammar.cpp.o
[  5%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/binary.cpp.o
[  6%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/diagnostic.cpp.o
[  6%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/disassemble.cpp.o
[  6%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/enum_string_mapping.cpp.o
[  7%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/ext_inst.cpp.o
[  7%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/extensions.cpp.o
[  7%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/libspirv.cpp.o
[  8%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/name_mapper.cpp.o
[  8%] Building CXX object source/CMakeFiles/SPIRV-Tools.dir/opcode.cpp.o
/build/source/source/opcode.cpp: In function 'int32_t spvOpcodeGeneratesType(SpvOp)':
/build/source/source/opcode.cpp:340:10: error: 'SpvOpTypeRayQueryProvisionalKHR' was not declared in this scope; did you mean 'SpvCapabilityRayQueryProvisionalKHR'?
  340 |     case SpvOpTypeRayQueryProvisionalKHR:
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          SpvCapabilityRayQueryProvisionalKHR
make[2]: *** [source/CMakeFiles/SPIRV-Tools.dir/build.make:243: source/CMakeFiles/SPIRV-Tools.dir/opcode.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:764: source/CMakeFiles/SPIRV-Tools.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
builder for '/nix/store/n0l83d4sgdaicyxfd8fmapxpxlxhbhic-spirv-tools-2020.2.drv' failed with exit code 2
error: build of '/nix/store/n0l83d4sgdaicyxfd8fmapxpxlxhbhic-spirv-tools-2020.2.drv' faile
```

**Comment by Ralith** [comment]
Ah, yeah, this probably has to be updated in lockstep with -tools.

**Pull request closed without merging**